### PR TITLE
Fix Variant types displayed as `Nil` in extension_api.json

### PR DESCRIPTION
**Pull request by touilleMan**
Related to https://github.com/godotengine/godot/pull/66774

### CODE REVIEW

File: binding_generator.py [hunk 1]
```diff
@@ -279,8 +279,7 @@ def generate_builtin_bindings(api, output_dir, build_config):
         if "operators" in builtin_api:
             for operator in builtin_api["operators"]:
                 if "right_type" in operator:
-                    # FIXME Temporary workaround for incorrect JSON
-                    if operator["right_type"] == "Nil":
+                    if operator["right_type"] == "Variant":
                         used_classes.add("Variant")
                     elif is_included(operator["right_type"], class_name):
```
Comment (akien-mga) [inline binding_generator.py:284]:
```suggestion
                    if is_included(operator["right_type"], class_name):
```
Probably works too?

Comment (touilleMan) [reply binding_generator.py]: done 👍

**Comment by akien-mga** [comment]
It seems to fail building despite syncing the json:
```
gen/src/variant/packed_color_array.cpp: In static member function 'static void godot::PackedColorArray::init_bindings()':
gen/src/variant/packed_color_array.cpp:82:173: error: 'GDNATIVE_VARIANT_TYPE_VARIANT' was not declared in this scope
  _method_bindings.operator_equal_Variant = internal::gdn_interface->variant_get_ptr_operator_evaluator(GDNATIVE_VARIANT_OP_EQUAL, GDNATIVE_VARIANT_TYPE_PACKED_COLOR_ARRAY, GDNATIVE_VARIANT_TYPE_VARIANT);
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by touilleMan** [comment]
@akien-mga I've updated the PR, now we have a green build ;-)

I'm waiting for a green review on this PR, then:
- I'll merge it
- I'll update https://github.com/godotengine/godot/pull/66774 to resume to master instead of this branch for godot-cpp in CI
- I'll wait for a green build and merge https://github.com/godotengine/godot/pull/66774

Sounds good ?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**